onnx/defs/shape_inference.h

```diff
@@ -312,12 +313,24 @@ inline void propagateShapeFromInputToOutput(
       ctx.getInputType(inputIndex)->tensor_type().shape();
 }
 
-inline void propagateShapeAndTypeFromFirstInput(InferenceContext& ctx) {
-  propagateElemTypeFromInputToOutput(ctx, 0, 0);
-  if (!hasNInputShapes(ctx, 1)) {
+inline void propagateShapeAndTypeFromInputToOutput(InferenceContext& ctx, size_t inputIndex, size_t outputIndex) {
+  propagateElemTypeFromInputToOutput(ctx, inputIndex, outputIndex);
+  if (!hasNInputShapes(ctx, inputIndex + 1)) {
```

Comment:
You should use `hasInputShape` here. Check how `hasNInputShapes` is implemented here: https://github.com/onnx/onnx/blob/master/onnx/defs/shape_inference.h#L266

Reply:
Thanks: 6aa269b